lib/Models/Filter.php

```diff
@@ -61,10 +61,11 @@ public function __construct($params)
             $filters = \json_decode($params['filters'], true);
             foreach($filters as $filter) {
                 if (in_array($filter['type'], self::$ALLOWED_FILTERS)) {
+                    $compare = $filter['compare'] ?? '!=';
                     $this->filters[] = [
                         'type'    => $filter['type'],
                         'value'   => preg_replace('/[^0-9a-zA-Z\w\ ]/', '', $filter['value']),
-                        'compare'  => $filter['compare'] == '=' ? '=' : '!='
+                        'compare'  => $compare
```

Comment:
```suggestion
                        'compare'  => $compare == '=' ? '=' : '!='
```

This prevents arbitrary compare operators from users, which could otherwise allow SQL injections. In the videos model, I couldn't identify a place where the operator is used directly in an SQL string, which could lead to an SQL injection.